app/src/main/java/com/bottlerocketstudios/brarchitecture/ui/repository/RepositoryBranchesViewModel.kt

```diff
@@ -0,0 +1,68 @@
+package com.bottlerocketstudios.brarchitecture.ui.repository
+
+import com.bottlerocketstudios.brarchitecture.R
+import com.bottlerocketstudios.brarchitecture.domain.models.Branch
+import com.bottlerocketstudios.brarchitecture.domain.repositories.BitbucketRepository
+import com.bottlerocketstudios.brarchitecture.ui.BaseViewModel
+import com.bottlerocketstudios.compose.repository.RepositoryBranchItemUiModel
+import com.bottlerocketstudios.compose.util.formattedUpdateTime
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.map
+import org.koin.core.component.inject
+import timber.log.Timber
+import java.time.Clock
+
+class RepositoryBranchesViewModel : BaseViewModel() {
+    // DI
+    private val repo: BitbucketRepository by inject()
+    private val clock by inject<Clock>()
+
+    // State
+    private val srcBranches = MutableStateFlow<List<Branch>>(emptyList())
+    val currentRepoName = MutableStateFlow("")
+
+    // Internal state
+    val repos = repo.repos.groundState(emptyList())
+
+    init {
+        launchIO {
+            currentRepoName.collect { repoName ->
+                getRepoBranches(repoName)
+            }
+        }
+    }
+
+    // UI
+    val path: StateFlow<String> = MutableStateFlow("")
+    val itemCount: StateFlow<Int> = srcBranches
+        .map { it.size }
+        .groundState(0)
+    val uiModels: StateFlow<List<RepositoryBranchItemUiModel>> = srcBranches
+        .map { branches ->
+            branches.map { branch ->
+                RepositoryBranchItemUiModel(
+                    name = branch.name,
+                    timeSinceCreated = branch.date.formattedUpdateTime(clock),
+                    status = "#1(OPEN)"
+                )
+            }
+        }
+        .groundState(emptyList())
+
+    // Load Logic
+    private fun getRepoBranches(name: String) {
+        val selectedRepo = repos.value.firstOrNull { it.name?.equals(name) ?: false }
+        path.setValue(name)
+        selectedRepo?.let {
+            val slug = it.workspace?.slug ?: ""
+            Timber.d(slug + " " + selectedRepo.name.orEmpty())
```

Comment:
@LuisRocketRangel nit: Remove this log OR modify it to include the function name as below:
```suggestion
            Timber.d("[getRepoBranches] $slug ${selectedRepo.name.orEmpty()})
```

Reply:
Removed log